### PR TITLE
codegen-kotlin: Generate toString() methods consistent with data classes

### LINKED ISSUE #717: codegen-kotlin: Use same toString() representation for data classes and regular classes
codegen-kotlin generates data classes whenever possible and regular classes otherwise.

For data classes, the default `toString()` implementation of data classes is used.
Example:
```
Person(name=Bob, address=Address(city=London))
`````

For regular classes, a `toString()` method is generated whose output somewhat resembles Pkl syntax.
Example:
```
Person {
  name = Bob
  address = Address {
    city = London
  }
}
```

It would be more consistent to use the same `toString()` representation for data classes and regular classes.
This could be achieved in one of the following ways:
1. Generate` toString()` methods for both data classes and regular classes.
2. Generate `toString()` methods for regular classes whose output matches that of data classes.

(1) enables a `toString()` representation that somewhat resembles Pkl syntax, whereas (2) generates much less code.

### PR DESCRIPTION
**Pull request by translatenix**
Note to reviewers: The second commit is a rename refactoring.

codegen-kotlin: Generate toString() methods consistent with data classes

Motivation:
codegen-kotlin generates a mix of data classes and regular classes.
For regular classes, toString() methods are also generated.
However, the output of generated toString() methods
differs from the output of default toString() methods of data classes.

Changes:
Generate toString() methods that produce the same output
as default toString() methods of data classes.

Result:
- Less and simpler code is generated.
- Object graphs containing a mix of data classes and regular classes are formatted consistently.
- Fixes #717.

### CODE REVIEW

File: pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt [hunk 1]
```diff
@@ -385,34 +345,25 @@ class KotlinCodeGenerator(
...
-          // to avoid making user code depend on kotlin-reflect
-          "builder.append(%T::class.java.simpleName).append(\" {\")",
-          kotlinPoetClassName
+          "return %P",
+          buildString {
+            append(pClass.simpleName)
+            append("(")
+            var isFirst = true
+            for (propertyName in allProperties.keys) {
+              if (isFirst) isFirst = false else append(", ")
```
Comment (bioball) [inline pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt:358]: [nit]: I think this is much easier to read:

```suggestion
              if (isFirst) {
                isFirst = false 
              } else {
                append(", ")
              }
```

File: pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt [hunk 2]
```diff
@@ -385,34 +345,25 @@ class KotlinCodeGenerator(
...
+              append(propertyName)
+              append("=$")
+              append(propertyName)
```
Comment (bioball) [inline pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt:361]: This needs to handle possibly quoted identifiers. Played around with it and this seems to do the trick:

```suggestion
              val propertyNameQuoted = CodeBlock.of("%N", propertyName).toString()
              if (propertyName == propertyNameQuoted) {
                append(propertyName)
              } else {
                append("{")
                append(propertyNameQuoted)
                append("}")
              }
```

**Review by bioball — CHANGES_REQUESTED** [review]
Thanks! Just one issue; we should handle quoted identifiers (see comment). Also, it would be good to have a test for it.

File: pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt (hunk 2 above)
Comment (translatenix) [reply pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt]: Good catch. I ended up using a `CodeBlock` for the entire interpolated string (as recommended in docs).

File: pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt (hunk 1 above)
Comment (translatenix) [reply pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt]: I replaced `isFirst` (here and in one other location) with a nicer construct.

**Comment by translatenix** [comment]
@bioball ready from my side

File: pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt [hunk 3]
```diff
@@ -344,24 +338,25 @@ class KotlinCodeGenerator(
...
-            append("(")
-            var isFirst = true
-            for (propertyName in allProperties.keys) {
-              if (isFirst) isFirst = false else append(", ")
-              append(propertyName)
-              append("=$")
-              append(propertyName)
+          CodeBlock.builder()
+            .apply {
+              add("%L", pClass.toKotlinPoetName().simpleName)
```
Comment (translatenix) [inline pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt:350]: added `toKotlinPoetName()` here, which I think is necessary

Comment (bioball) [reply pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt:350]: Ah, yup; good catch

**Pull request merged**